backend/src/main/java/com/google/sps/data/Participant.java

```diff
@@ -3,46 +3,41 @@
 import com.google.common.base.Preconditions;
 import java.time.ZonedDateTime;
 
-/** A user who want to be matched. */
+/** A user who wants to be matched. */
 public final class Participant {
 
-  /** Datastore ID */
-  private final long id;
   /** Google username (ldap) */
   private final String username;
-  /** Time user is starts being available */
+  /** Time user starts being available */
   private final ZonedDateTime startTimeAvailable;
   /** Time user is available until */
   private final ZonedDateTime endTimeAvailable;
   /** How long user wants to chat */
   private final int duration;
+  /** Current match in datastore, null if never found a match or already been returned */
+  private final long currentMatchId;
   /** Time of submitted form */
   private final long timestamp;
 
   /** Initialize constructor fields */
   public Participant(
-      long id,
       String username,
       ZonedDateTime startTimeAvailable,
       ZonedDateTime endTimeAvailable,
       int duration,
+      long currentMatchId,
       long timestamp) {
```

Comment:
Once a Participant has a match, should they be removed from the system? If so, why add a field for currentMatchId?

Reply:
Once a Participant has a match, their availability fields and currentMatchId are nulled out, but they still remain in the datastore. We decided not to remove participants from the datastore since participants have the option to save their preferences. If they choose to save their preferences, their availability and matchId will be nulled out each time their match is successfully returned to them, but their other personal preferences (role, PA, interests) will be saved. If they choose NOT to save their preferences, we still keep them in the datastore as a participant but null out ALL fields. (Right now, the only fields are time availability and matchId).

Reply:
This sounds to me like a conflation of two data models that could be separated out in your datastore. One is a `User` model that stores any preferences the user opted to save and the other is a `Participant` model which represents a user actively looking for a match. Otherwise you risk Participant becoming like a Union class where it either represents someone looking for a match (fields not null) or just stored user information (nulled out match fields). WDYT?

Reply:
I see. In our pod meeting today, we talked about this and decided to make this a new PR after this one.

Reply:
Would it make sense to author and submit the UserDatastore PR first?